src/test/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/MessageTest.java

```diff
@@ -50,12 +55,253 @@ void testFindFirstGpCodeDefaultsTo9999() {
     @Test
     void testFindFirstGpCodeReturnsCorrectly() {
         final Message msg = new Message(List.of(
-            "NAD+GP+2750922,295:900'",
-            "NAD+GP+1649811,184:899'"
+            "NAD+GP+2750922,295:900",
+            "NAD+GP+1649811,184:899"
         ));
 
         String firstGpCode = msg.findFirstGpCode();
 
         assertEquals("2750922,295", firstGpCode);
     }
+
+    @Test
+    void testSegmentGroups() {
```

Comment:
This method tests a full(?) example of a single message. Kinda shows how the data is extracted, and tries to cover all the parts of the input segments. Only here to demonstrate that the approach is working - not intended to be used for a PR to merge to main.